test/unit/utils.authentication.unit.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect, toJSON} from '@loopback/testlab';
+import {MongoDataSource} from '../../src/datasources';
+import {
+  decodeAccessToken,
+  getAccessTokenForUser,
+} from '../../src/utils/user.authentication';
+import {UserRepository, OrderRepository} from '../../src/repositories';
+import {User} from '../../src/models';
+import * as _ from 'lodash';
+const SECRET = 'secretforjwt';
+
+describe('authentication', () => {
+  const mongodbDS = new MongoDataSource();
+  const orderRepo = new OrderRepository(mongodbDS);
+  const userRepo = new UserRepository(mongodbDS, orderRepo);
+  const user = {
+    email: 'unittest@loopback.io',
+    password: 'p4ssw0rd',
+    firstname: 'unit',
+    surname: 'test',
+  };
+  let newUser: User;
+
+  before('create user', async () => {
+    newUser = await userRepo.create(user);
+  });
+
+  it('decodes valid access token', async () => {
+    const token = await getAccessTokenForUser(userRepo, {
+      email: 'unittest@loopback.io',
+      password: 'p4ssw0rd',
+    });
+    const expectedUser = getExpectedUser(newUser);
+    const currentUser = await decodeAccessToken(token, SECRET);
+    expect(currentUser).to.deepEqual(expectedUser);
+  });
+
+  it('throws error for invalid accesstoken', async () => {
+    const token = 'fake';
+    try {
+      await decodeAccessToken(token, SECRET);
+      expect('throws error').to.be.true();
```

Comment:
@jannyHou 

Here is a better way for verifying that a call of an async function throws an error:

```ts
await expect(decodeAccessToken(token, SECRET))
  .to.be.rejectedWith('jwt malformed');
```

Reply:
@bajtos Thanks! I will address the feedback in https://github.com/strongloop/loopback4-example-shopping/pull/37